torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2535,6 +2535,34 @@ def sample_inputs_kron(op_info, device, dtype, requires_grad):
         sample_inputs.append(sample)
     return tuple(sample_inputs)
 
+def sample_inputs_scatter_ops(op_info, device, dtype, requires_grad):
+    def _make(shape, dtype=dtype, low=None, high=None):
+        return make_tensor(shape, device, dtype, low=low, high=high, requires_grad=requires_grad)
+
+    test_cases = (
+        ((M, S), (S, S)),
+        ((M, S), (M, S // 2)),
+    )
+
+    samples = []
+    for self_shape, shape in test_cases:
+        for dim in [0, 1, -1]:
+            other_dim = 1 - (dim if dim >= 0 else -dim)
+            index = gather_variable(shape, other_dim, self_shape[dim], device=device)
+            samples.append(SampleInput(_make(self_shape), args=(dim, index, _make(shape))))
+
+    samples.append(
+        SampleInput(_make(()), args=(0, torch.tensor(0, dtype=torch.long, device=device), _make(()))),
+    )
+
+    if op_info.name == 'scatter':
```

Comment:
scatter_add should also be tested for scalar input

Reply:
I don't think `scatter_add` supports scalar inputs. Am I missing something?
https://github.com/pytorch/pytorch/blob/0a541e23e17a2b9555673c0877e15abcfe49a173/aten/src/ATen/native/native_functions.yaml#L4858-L4867